src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CMDLineProgramModule.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+/**
+ * A class for experimenting with cmd line programs (such as bwa and sga) that's not part of GATK (or has Java bindings yet).
+ */
+public abstract class CMDLineProgramModule {
+
+    public CMDLineProgramModule(){  }
+
+    /**
+     * Builds the initial argument list for the command line (e.g. "bwa mem").
+     * @return a list of strings having the names of requested tools.
+     * @throws IOException
+     */
+    abstract ArrayList<String> initializeCommands() throws IOException;
+
+    /**
+     * Worker function to call the cmd line program.
+     * @param runtimeArguments          arguments to be provided to the program
+     * @param directoryToWorkIn         a directory for the program to work in (mainly for output)
+     * @param stdoutDestination         redirects the stdout from the program to a file
+     * @param stderrMessageDestination  appends the stderr message to a string, which is assumed to be not null
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws RuntimeException
+     */
+    public void run(final ArrayList<String> runtimeArguments,
+                    final File directoryToWorkIn,
+                    final File stdoutDestination,
+                    String stderrMessageDestination) throws IOException, InterruptedException, RuntimeException{
+
+        final ArrayList<String> commands = initializeCommands();
+        for(final String arg : runtimeArguments){ commands.add(arg); }
+
+        final ProcessBuilder builder = new ProcessBuilder(commands);
+        setupWorkingDir(builder, directoryToWorkIn, stdoutDestination);
+        setupWorkingEnvironment(builder);
+
+        Process runProcess = builder.start();
+        int exitStatus = runProcess.waitFor();
+        stderrMessageDestination += redirectSTDERR(runProcess);
+
+        if(0!=exitStatus){ onError(commands, stderrMessageDestination, exitStatus); }
+    }
+
+    protected void setupWorkingDir(final ProcessBuilder builder,
+                                   final File directoryToWorkIn,
+                                   final File stdoutDestination){
+        builder.directory(directoryToWorkIn);
+        if(null!=stdoutDestination) { builder.redirectOutput(stdoutDestination); }
```

Comment:
Are you assuming that the stdoutDestination will be in the working directory (since you have these in a method together?) Might add a comment to that effect if so, or just inline this method since these lines don't seem too tied together at first glance.